app/lib/features/chat/widgets/room_avatar.dart

```diff
@@ -38,17 +38,17 @@ class RoomAvatar extends ConsumerWidget {
       data: (profile) {
         //Show conversations avatar if available
         //Group : Show default image if avatar is not available
-        if (profile.hasAvatar() || !profile.isDm) {
+        if (!convo.isDm()) {
           return ActerAvatar(
             uniqueId: roomId,
-            mode: profile.isDm ? DisplayMode.User : DisplayMode.Space,
+            mode: DisplayMode.Space,
             displayName: profile.displayName ?? roomId,
             avatar: profile.getAvatarImage(),
             size: 18,
           );
         }
```

Comment:
but a DM that has an Avatar _also_ should be showing that avatar ... that's why the previous check was right and the current one wouldn't be, no?

```suggestion
        } else if (profile.hasAvatar()) {
                  return ActerAvatar(
            uniqueId: roomId,
            mode: DisplayMode.User,
            displayName: profile.displayName ?? roomId,
            avatar: profile.getAvatarImage(),
            size: 18,
          );
          }
        
```

Reply:
> We are handling that case here.  

[where](https://github.com/acterglobal/a3/pull/1103/files#diff-4015547fb05cb94d3fa13a84efb72ab7c8fc980c44c4e6f8ff2b65cc5d34c249R70-R97)? `dmAvatar` does not check whether the chat has an Avatar set, it uses the _members_ Avatars regardless ...

>  Also profile.hasAvatar() don't seem useful here as ActerAvatar would return fallback if avatar is not present.

I don't get that comment. `profile` is the `ConvoProfile` here. If that chat has an avatar, regardless of whether it is a DM or not, we want to show _that_ avatar. That is all this is about. Not the fallback or anything else.

The fallback is different depending on `isDm == true` or not, but if we have an avatar for the room, we _always_ want to show the configured avatar. isDm only decides whether we use a circle or the square ...

Reply:
Okay seems reasonable. I will address it.